raft/node.go

```diff
@@ -314,13 +319,18 @@ func (n *node) run(r *raft) {
 		// TODO: maybe buffer the config propose if there exists one (the way
 		// described in raft dissertation)
 		// Currently it is dropped in Step silently.
-		case m := <-propc:
+		case pm := <-propc:
+			m := pm.m
 			m.From = r.id
-			r.Step(m)
+			err := r.Step(m)
+			if pm.result != nil {
+				pm.result <- err
+				close(pm.result)
+			}
 		case m := <-n.recvc:
 			// filter out response message from unknown From.
 			if pr := r.getProgress(m.From); pr != nil || !IsResponseMsg(m.Type) {
-				r.Step(m) // raft never returns an error
+				r.Step(m)
```

Comment:
do we consume the returned error?

Reply:
As you comment before:

>do we really need to change the recvc chan? if the proposal is redirected from other peers, the waiter is actually on another node anyway. there is no easy to propagate back the error even with this change.
probably changing the propc is good enough for the current purpose.

I leaved this unchanged